Lean4Axiomatic/Natural/Exponentiation.lean

```diff
@@ -32,26 +33,28 @@ instance (priority := default+1) pow_inst
   pow := Exponentiation.Ops._pow
 }
 
-/-- Properties of exponentiation to a natural number. -/
+/-- Properties of exponentiation for a monoid type α to a natural number. -/
 class Exponentiation.Props
-    {α : Type} {ℕ : outParam Type} (mul : outParam (α → α → α))
-    [Core ℕ] [Addition ℕ] [Multiplication ℕ] [EqvOp α] [OfNat α 1] [Ops α ℕ]
+    {α : Type} {ℕ : outParam Type}
+    [Core ℕ] [Addition ℕ] [Multiplication ℕ] [EqvOp α] [Ops α ℕ]
+    [isMonoid : CA.Monoid.Monoid α]
     :=
-  /-- Any number raised to the power zero, is one. -/
-  pow_zero {x : α} : x ^ (0:ℕ) ≃ 1
-
+  /- TODO: replace requirement of being a monoid with being a commutative ring. -/
```

Comment:
You're suggesting this TODO because many of the theorems below take `AA.Commutative` as arguments? I think it's okay to stick with just a monoid. Not all of the theorems require commutativity; if we required `α` to be a commutative ring, then we would no longer be able to use those theorems with monoids.